[MSITE-1010] Add documentation how site repository inheritance

Bukama opened the pull request:
Proposed commit message:

```
[MSITE-1010] Add documentation how site repository inheritance

More information including an example how the site repository configuration is inherited,
when using a parent-pom.

closes #MSITE-1010

```

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MSITE) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x]  Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MSITE-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSITE-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

michael-o commented on src/site/apt/examples/inheritancesiterepository.apt:
```diff
@@ -0,0 +1,77 @@
+  for the site-repository for multiple projects, as otherwise files will overwrite files of the first deployed project
+  are overwritten by the files of the second deployed one.
+
+  To handle this easily you can use a the same parent-pom (often called "meta-pom") for each of your project.
```
Remove the term meta POM. It will cause confusion with the reader.

Bukama commented:
Thanks for the review. Requested changes done.

michael-o reviewed (changes requested):
Yet another issue. The problem is not a repository, we do not even have that notion for sites. It is solely about site deployment vs local generation -- as laid out in the JIRA issue. We shouldn't try to introduce the term site repository. I'll review our code to make wording better.

michael-o commented:
See: https://github.com/apache/maven-site-plugin/pull/195

Bukama commented:
> Yet another issue. The problem is not a repository, we do not even have that notion for sites. It is solely about site deployment vs local generation -- as laid out in the JIRA issue. We shouldn't try to introduce the term site repository. I'll review our code to make wording better.

I get your point and totally understand it. I updated the PR to avoid the term repository, changed the name of the page (and therefore the menu entry) to distribution management and also fixed some grammar mistakes/bad wording. Would you please have another look?

michael-o commented:
Will review tomorrow again.

michael-o commented:
@Bukama I have added a few improvements, please have a look.

Bukama commented:
> @Bukama I have added a few improvements, please have a look.

Thanks! I like them!

jira-importer commented:
Resolve #1141
